### ISSUE TITLE
container finished log does not include pod name - this makes it difficult to identify out of memory kills

### ISSUE THREAD

**Issue by allen-ziegenfus** [kubernetes/kubernetes#125785]
### What would you like to be added?

When a container dies due to OOMKill, there are these types of logs generated:

kubelet.go:2447] "SyncLoop (PLEG): event for pod" pod="<namespace/pod template hash>"

generic.go:334] "Generic (PLEG): container finished" podID="<podid....>" containerID="<containerid...>" exitCode=137

Would it be possible to consistently log the pod name (namespace/template hash) in the container finished log message (https://github.com/kubernetes/kubernetes/blob/57645002432c52aa743e69739613ca337616441a/pkg/kubelet/pleg/generic.go#L334) ?

From the definition of the Pod struct it appears that this information would also be available in generic.go

### Why is this needed?

This would make the log messages more consistent and make it easier to find OOM kills by pod name. Currently it involves having to find the podID which makes it difficult to find these messages.

**Comment by vunnamkowsik** [comment]
Yes you can change that line with adding pod.name. Before that we need to have permission for pod object and check the variable name is available or not.

**Comment by ffromani** [comment]
/triage accepted
/priority backlog

I think the best way to fix this issue is porting more of the kubelet towards structured/contextual logging. While this is not strictly a prerequisite to fix, it will greatly help to have consistent logs. I'll add a few links shortly

**Comment by allen-ziegenfus** [comment]
@ffromani  I agree! Another idea I had - would it be also possible to publish the OOM condition as an Event so it's more visible? I am new to this so not sure where ideas like this would be discussed.

**Comment by allen-ziegenfus** [comment]
@ffromani  - Good thoughts!

Admittedly my experience / motivation regarding this comes from the question of ease of use when interpreting GCP logs - so it's partly a downstream question of how GCP interprets and parses things. Events are logged in GCP with their context which is filterable (e.g. I can easily find pod specific logs quickly) which is one reason I was mentioning that.

These kind of unstructured kubelet logs are not easily filterable  - I will have to check for an example where structured logging is used to see how those are processed.

But with this change we can at least more easily find the pod exit code when searching for a given pod name, that's what I was after - right now I have to look up the podID as an intermediate step.
